src/icon/icon.ts

```diff
@@ -0,0 +1,64 @@
+import { WidgetBase } from '@dojo/widget-core/WidgetBase';
+import { DNode } from '@dojo/widget-core/interfaces';
+import { ThemedMixin, ThemedProperties, theme } from '@dojo/widget-core/mixins/Themed';
+import { v } from '@dojo/widget-core/d';
+import { CustomAriaProperties } from '../common/interfaces';
+import { formatAriaProperties } from '../common/util';
+import * as iconCss from '../theme/common/icons.m.css';
+import { customElement } from '@dojo/widget-core/decorators/customElement';
+
+export type IconType = keyof typeof iconCss;
+
+/**
+ * @type IconProperties
+ *
+ * Properties that can be set on an Icon component
+ *
+ * @property type           Icon type, e.g. downIcon, searchIcon, etc.
+ */
+
+export interface IconProperties extends ThemedProperties, CustomAriaProperties {
+	type?: IconType;
+	onClick?(): void;
+}
+
+export const ThemedBase = ThemedMixin(WidgetBase);
+
+@theme(iconCss)
+@customElement<IconProperties>({
+	tag: 'dojo-icon',
+	properties: [
+		'theme',
+		'aria',
+		'extraClasses'
+	],
+	attributes: [ 'type' ],
+	events: [
+		'onClick'
+	]
+})
+export class IconBase<P extends IconProperties = IconProperties> extends ThemedBase<P, null> {
+	private _onClick (event: MouseEvent) {
+		event && event.stopPropagation();
+		this.properties.onClick && this.properties.onClick();
+	}
+
+	constructor() {
+		super();
+	}
+
+	protected render(): DNode {
+		const {
+			aria = {},
+			type = 'downIcon'
```

Comment:
I don't think we should have a default for `type`

Reply:
Should the property be required on the props then or just render without an icon type if it's not provided?